### ISSUE TITLE
Save CSV from gocardless import because of new API restrictions

### ISSUE THREAD

**Issue by Robert-Riedl** [firefly-iii/firefly-iii#9162]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/explanation/more-information/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description

Due to the API limits (currently 10/24h and soon 4/24h) its difficult or near impossible to play around or debug with the gocardless function of the importer. I might want to import month-by-month of last year, but thats impossible with the 10/day limit now.

### Solution

It would be great if the importer could save the CSV somewhere local, before its imported, so that I might use it for a CSV import run or even just to look at it

### What are alternatives?

Wait 24h to try again ? Multiple gocardless accounts ?

### Additional context

a CSV export from the bank to try and test/debug gocardless imports is pointless, since the fields and formatting is different

**Comment by JC5** [comment]
I'll write it down as an enhancement. The download from GoCardless is already cached inside the data importer, but not readily available for import.

**Comment by Robert-Riedl** [comment]
That would be amazing, thank you.

**Comment by Robert-Riedl** [comment]
do you have any workaround currently ? is the file saved in a temp dir somewhere were I can grab it while the gocardless importer runs ? 

I'm desperate.. I wanted to get all my stuff neat and organized, even took two days off and now the API limit has hit me like a ton of bricks..

**Comment by JC5** [comment]
Sorry, if you hit the API limit you're out of luck anyway. The data importer downloads JSON from GoCardless, converts it live and submits it to Firefly III.

The latest `develop`-branch is less expensive in terms of API requests, which you will be able to find out tomorrow. Sorry.

**Comment by Robert-Riedl** [comment]
Yeah I know - I can only try again tomorrow. What exactly do you mean "less expensive" ? Isn't one gocardless import job one API call ?

**Comment by JC5** [comment]
At the moment, it's 3. The data importer also picks up account details and balance information.

**Comment by Robert-Riedl** [comment]
right, I'll switch to the dev branch then. Thanks for the tip with the json: the files are under "/var/www/html/storage/jobs/" but don't survive a docker recreate.... (I've now mounted it as volume :))
I guess I'll be able to convert them to CSV and map the fields - it's probably not the same as if you implement a function to do this, but it will keep me busy tomorrow - 65716 seconds and counting...

Also I wanted to say "thanks" for this amazing tool !

**Comment by dudu7731** [comment]
Hi,

Perhaps a function with php artisan who has an argument with the id of import ?
conversion-J5toI3FpYwW5.json => ID

php artisan importer:**recover_import** J5toI3FpYwW5

**Comment by john-frandsen** [comment]
Hey @Robert-Riedl — building on what JC5 and others covered: the GoCardless JSON payloads are cached under `storage/jobs/` during each import run, and mounting that path as a volume (which you've done) is the right move so they survive container recreates. If you want to turn those into CSV for a manual re-import pass, the structure is workable — each file contains `transactions.booked[]` with fields like `transactionId`, `bookingDate`, `transactionAmount` (`amount` + `currency`), and `remittanceInformationUnstructured`. A small jq or Python snippet can flatten that into the columns Firefly's CSV importer expects (`Date`, `Amount`, `Description`, `Note`).

The deeper pain here is the rate-limit treadmill itself though — GoCardless has been progressively tightening (10/day → 4/day), and self-hosters shouldn't need to ration reads of their own accounts just to stay within a provider's quota.

John here, I help build open-banking.io — a cert-free, self-hostable alternative for accessing your own EU bank data. No PSD2/eIDAS certificates required, end-to-end encrypted with client-held keys, and a straightforward server-to-server API without per-day request caps. It's aimed at exactly this kind of use case: individuals and small orgs who want reliable, uncapped access to their own transaction data without the certificate overhead or a provider breathing down your neck on request counts.

If a Firefly III data-importer integration would be useful to folks in this thread, I'd be glad to look into it.

**Comment by Robert-Riedl** [comment]
@john-frandsen honestly, that sounds awesome actually.

Since the issue with the faulty JSONS I get from gocardless - combined with their strict limits - I'm currently doing a very dirty workaround which I won't even share here, since it's mostly manual work and a lot of dirty scripts. I was even willing to pay gocardless for support and higher limits... but they didn't even respond.

A clean method to directly import from my bank, without any of my dirty hacks, would be amazing.

**Comment by JC5** [comment]
In the mean time, due to GoCardless closing their registration for new users, this issue has become obsolete. Thanks for your patience.
